<a href="https://colab.research.google.com/github/Omkar-Udawant/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!git clone https://github.com/Omkar-Udawant/flyrank-ml-internship.git
%cd flyrank-ml-internship

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 123, done.
remote: Counting objects: 100% (123/123), done.
remote: Compressing objects: 100% (76/76), done.
remote: Total 123 (delta 38), reused 99 (delta 31), pack-reused 0 (from 0)
Receiving objects: 100% (123/123), 1.85 MiB | 2.96 MiB/s, done.
Resolving deltas: 100% (38/38), done.
/content/flyrank-ml-internship/flyrank-ml-internship


In [ ]:
!pwd

/content/flyrank-ml-internship/flyrank-ml-internship


In [ ]:
!ls data/raw

content_refresh_anonymized.csv


# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Omkar-Udawant/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# 1. My Lane (or freestyle) and why

**Selected Lane:** Refresh / Content Opportunity Scoring

I selected the **Refresh / Content Opportunity Scoring** lane because it focuses on identifying which content pages should be reviewed first using observable search and engagement signals. The objective is to prioritize pages that may benefit from actions such as refreshing, expanding, monitoring, or pruning. I chose this lane because it combines data analysis with machine learning to support better decision-making while keeping the recommendations transparent and interpretable.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# 2. The Question

## Research Question
Which content pages should be prioritized for review based on search performance and engagement signals so that the content team can focus on pages with the highest opportunity for improvement?

## Decision
Decide which pages should be reviewed first for actions such as content refresh, expansion, monitoring, or pruning.

## Unit of Analysis
One content page.

## Output
A ranked list of content pages with a priority score indicating which pages deserve review first.

## Action
The SEO or content team can review the highest-ranked pages and decide whether to refresh content, improve metadata, expand information, monitor performance, or remove outdated content.

## Cost of a Wrong Recommendation
If the wrong pages are prioritized, the team may spend time and resources improving pages with little potential while missing pages that have a greater opportunity for improvement. This can reduce the overall effectiveness of content optimization.

## Why Data and Machine Learning?
Search performance depends on several observable factors such as impressions, clicks, click-through rate (CTR), average position, content age, and engagement metrics. Machine learning can combine these signals to identify patterns and rank pages more effectively than fixed rules alone. The goal is to support human decision-making, not replace it.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Display key statistics
print("Number of Rows:", len(df))
print("Number of Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(), 4))
print("Average Position:", round(df["avg_position"].mean(), 2))

print("\nContent Type Distribution:")
print(df["content_type"].value_counts())

Number of Rows: 30000
Number of Columns: 44
Average CTR: 0.5107
Average Position: 16.34

Content Type Distribution:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64


# 3. Quick Look at the Data

To understand whether this research lane is worth exploring, I examined a few key statistics from the starter dataset.

The dataset contains **30,000 content pages** with **44 features**, providing a strong foundation for exploratory analysis and machine learning.

The average **click-through rate (CTR)** is **0.5107**, while the **average search position** is **16.34**, indicating variation in search visibility and user engagement across different pages.

The most common content type is **keyword article**, with **27,207 pages**, suggesting that a large portion of the dataset consists of content that can be evaluated for refresh opportunities.

These observations suggest that the dataset contains sufficient variation in search performance and content characteristics to support the **Refresh / Content Opportunity Scoring** lane.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# 4. Careful words: what I can and can't claim

This analysis is based on the anonymized starter dataset provided for the internship. The observations identify patterns and relationships in the data, but they do not prove that one factor directly causes another.

The ranked recommendations produced by this project are intended to support human decision-making. They should be reviewed alongside domain knowledge and validated with additional data before being used in practice.

This project does not claim to prove Google's ranking algorithm or guarantee that refreshing a page will improve its performance. It provides evidence-based insights using the available search and engagement signals.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


# 5. Self-check

- ✅ I selected the **Refresh / Content Opportunity Scoring** lane.
- ✅ I clearly defined my research question.
- ✅ I identified the decision, unit of analysis, output, action, and cost of an incorrect recommendation.
- ✅ I included real numbers from the starter dataset to support my lane choice.
- ✅ I used careful language and avoided making causal claims.
- ✅ My notebook is executed and ready to be committed to my GitHub repository.